### PR TITLE
CDPT-1034 CT Rejections: Add Rejected to title

### PR DESCRIPTION
**Pull request by MeenaModhvadia**
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### CODE REVIEW

File: app/views/cases/_new.html.slim [hunk 1]
```diff
@@ -2,7 +2,7 @@
   = t('.heading')
 
 - content_for :sub_heading
-  = t4c(kase, '', 'sub_heading', case_type: kase.type_abbreviation)
+  = get_offender_sar_heading(@offender_sar_type, @case)
```
Comment (vertism) [inline app/views/cases/_new.html.slim:5]: This needs to be more generic so it works for any case type. At the moment it's going to break the new case page for FOIs and SARs.

To start you should create a method that takes `kase` as a param and then have it output the exact same thing it does now.

Then you need to work out how to determine when the new case is a rejected offender sar based on information in that kase variable.

File: app/views/cases/new.html.slim [hunk 2]
```diff
@@ -2,7 +2,7 @@
   = t('.heading')
 
 - content_for :sub_heading
-  = t4c(@case, '', 'sub_heading', case_type: @case.pretty_type)
+  - get_offender_sar_heading(@offender_sar_type, @case)
```
Comment (vertism) [inline app/views/cases/new.html.slim:5]: Does this need to change?

File: app/views/cases/_new.html.slim (hunk 1 above)
Comment (vertism) [reply app/views/cases/_new.html.slim:5]: It is also possible that you can made changes to the `TranslateForCase` class (which is what the t4c method eventually calls to account for a rejected offender sar there.

File: config/locales/filters.en.yml [hunk 3]
```diff
@@ -89,14 +89,15 @@ en:
...
-      not-high-profile: Not high profile 
+      high-profile: High profile
+      not-high-profile: Not high profile
     filter_partial_case_flag:
       partial-case: Is partial case
       not-partial-case: Is not partial case
     filter_status:
       open: 'Open'
       closed: 'Closed'
+      rejected: 'Rejected'
```
Comment (vertism) [inline config/locales/filters.en.yml:100]: I'm not sure this change is required for this ticket.

File: config/locales/page_titles.en.yml [hunk 4]
```diff
@@ -19,6 +19,7 @@ en:
       ico: Creating an ICO case - Track-a-query
       offender_sar:
         subject_details_step: Subject details - Creating an Offender SAR case - Track-a-query
+        rejected_details_step:  Subject details - Creating a Rejected Offender SAR case - Track-a-query
```
Comment (vertism) [inline config/locales/page_titles.en.yml:22]: I'm not sure this addition makes sense.

File: config/locales/state.en.yml [hunk 5]
```diff
@@ -15,6 +15,7 @@ en:
       ready_to_dispatch: Ready to dispatch
       vetting_in_progress: Vetting in progress
       waiting_for_data: Waiting for data
+      rejected: Rejected
```
Comment (vertism) [inline config/locales/state.en.yml:18]: This may relate to a different ticket

File: config/routes.rb [hunk 6]
```diff
@@ -163,6 +163,7 @@
       get "deleted" => "filters#deleted"
       get "incoming" => "filters#incoming"
       get "/" => "filters#show"
+      get "rejected" => "filters#rejected"
```
Comment (vertism) [inline config/routes.rb:166]: I don't think this is necessary for this ticket

File: config/state_machine/configs/00_moj/20_sar/20_sar_standard_workflow.yml [hunk 7]
```diff
@@ -109,6 +109,10 @@
                   if: Workflows::Predicates#responder_is_member_of_assigned_team?
                   after_transition: Workflows::Hooks#reassign_user_email
 
+              rejected:
```
Comment (vertism) [inline config/state_machine/configs/00_moj/20_sar/20_sar_standard_workflow.yml:112]: This doesn't seem relevant for this ticket.
This also goes for the other changes to the state machine config.

File: app/views/cases/_new.html.slim (hunk 1 above)
Comment (MeenaModhvadia) [reply app/views/cases/_new.html.slim]: I am thinking of adding a flag to set on the case if it's a rejected offender sar then the t4c can check and output the rejected heading if its an offender sar & rejected.
What do you think?

Comment (vertism) [reply app/views/cases/_new.html.slim]: That seems like a reasonable idea.

**Comment by MeenaModhvadia** [comment]
This pr is no longer needed as created new branch off cdpt-1033 to allow for those changes as my ticket follows on from there

**Pull request closed without merging**